src/server/transaction.cc

```diff
@@ -398,13 +398,14 @@ bool Transaction::RunInShard(EngineShard* shard) {
         }
       }
       sd.local_mask &= ~OUT_OF_ORDER;
-
-      // It has 2 responsibilities.
-      // 1: to go over potential wakened keys, verify them and activate watch queues.
-      // 2: if this transaction was notified and finished running - to remove it from the head
-      //    of the queue and notify the next one.
-      if (shard->blocking_controller())
-        shard->blocking_controller()->RunStep(awaked_prerun ? this : nullptr);
+    }
+    // It has 2 responsibilities.
+    // 1: to go over potential wakened keys, verify them and activate watch queues.
+    // 2: if this transaction was notified and finished running - to remove it from the head
+    //    of the queue and notify the next one.
+    // RunStep is also called for global transactions because of commands like MOVE.
+    if (shard->blocking_controller()) {
+      shard->blocking_controller()->RunStep(awaked_prerun ? this : nullptr);
     }
```

Comment:
I had to move this out of the `!IsGlobal()` if-branch. Should this be called on every global transaction? Or should we distinguish MOVE with some flag. However, I guess in most cases it will be an empty run, so it'll have little affect 🤔

Reply:
yes, global transactions are rare and slower than regular ones. another function call is fine.